main.py

```diff
@@ -0,0 +1,140 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import glob
+import time
+
+from archive_converter import ArchiveMessageIndex
+from archive_updater import fill_message_directory
+from gerrit import upload_all_comments, get_gerrit_rest_api, Gerrit, find_and_label_all_revision_ids
+from git import GerritGit
+from message_dao import MessageDao
+from patch_parser import map_comments_to_gerrit, parse_comments, Patch, Patchset
+from setup_gmail import Message
+from typing import List, Set, Tuple
+
+GIT_PATH = '../linux-kselftest/git/0.git'
+FILE_DIR = 'index_files'
+GERRIT_URL = 'https://linux-review.googlesource.com'
+GOB_URL = 'http://linux.googlesource.com'
+COOKIE_JAR_PATH = 'gerritcookies'
+WAIT_TIME = 10
+
+class Server(object):
+    def __init__(self):
+        self.gerrit = Gerrit(get_gerrit_rest_api(COOKIE_JAR_PATH, GERRIT_URL))
+        self.gerrit_git = GerritGit(git_dir='gerrit_git_dir',
+                               cookie_jar_path=COOKIE_JAR_PATH,
+                               url=GOB_URL, project='linux/kernel/git/torvalds/linux', branch='master')
+        self.message_dao = MessageDao()
+        self.archive_index = ArchiveMessageIndex(self.message_dao)
+        self.last_hash = self.message_dao.get_last_hash()
+    
+    @staticmethod
+    def remove_files(file_dir : str):
+        files = glob.glob(f'{file_dir}/*')
+        for f in files:
+            os.remove(f)
+    
+    @staticmethod
+    def split_parent_and_reply_messages(messages : List[Message]) -> Tuple[List[Message], List[Message]]:
+        ''' Splits a list of messages into parent (first email in a thread) and replies.
+        Returns a list of parents, then replies '''
+        parents : List[Message] = []
+        replies : List[Message] = []
+        for message in messages:
+                if not message.in_reply_to:
+                    parents.append(message)
+                    continue
+                replies.append(message)
+        return (parents, replies)
+    
+    def update_message_dir(self) -> List[Message]:
+        self.last_hash = fill_message_directory(GIT_PATH, FILE_DIR, self.last_hash)
+        messages = self.archive_index.update(FILE_DIR) 
+        return messages
+    
+    def upload_messages(self, messages_to_upload : List[str]):
+        for message_id in messages_to_upload:
+            try:
+                email_thread = self.archive_index.find(message_id)
+                patchset = parse_comments(email_thread)
+                self.gerrit_git.apply_patchset_and_cleanup(patchset, self.message_dao)
+                find_and_label_all_revision_ids(self.gerrit, patchset)
+                upload_all_comments(self.gerrit, patchset)
+            except ValueError:
```

Comment:
optional: we should probably introduce a more specific error type here instead of overloading a bulitin one like `ValueError`

Doesn't have to happen in this PR necessarily.
Can consider adding a todo
```python
except ValueError:  # TODO(willliu@google.com): use a more specific exception
```